src/vs/workbench/contrib/terminal/electron-browser/terminalInstance.ts

```diff
@@ -739,14 +739,33 @@ export class TerminalInstance implements ITerminalInstance {
 		}
 		if (this._processManager) {
 			this._processManager.dispose(immediate);
+		} else {
+			// In cases where there is no associated process (for example executing an exetnsion callback task)
+			// consumers still expect on onExit event to be fired. An example of this is terminating the extnesion callback
+			// task.
+			this._onExit.fire(this._id);
```

Comment:
onExit fires with the exit code, not the ID

Reply:
Nice catch. This should be firing undefined at this point.